src/firefly/js/util/expr/Variable.js

```diff
@@ -6,20 +6,23 @@
 const variables = new Map();
 
 /**
-  * Return a unique variable named `name'.  There can be only one
-  * variable with the same name returned by this method; that is,
-  * makeVariable(s1) === makeVariable(s2) if and only if s1===s2.
-  * @param {String} name the variable's name
-  * @return the variable; create it initialized to 0 if it doesn't
-  *         yet exist */
+ * Return a unique variable named `name'.  There can be only one
+ * variable with the same name returned by this method; that is,
+ * makeVariable(s1) === makeVariable(s2) if and only if s1===s2.
+ * @param {String} name the variable's name
+ * @return the variable; create it initialized to 0 if it doesn't
+ *         yet exist */
 export function makeVariable(name) {
-     let result = variables.get(name);
-     if (!result) {
-         result = new Variable(name);
-         variables.set(name, result);
-     }
-     return result;
- }
+    if (!/^[A-Za-z\d_]+$/.test(name) && !name.startsWith('"')) {
+        name = `"${name}"`;
+    }
```

Comment:
I think you should create a function `quoteNonAlphaNum()`.. or something like it in ChartUtil to put his logic in there.  It's not that the logic is hard, it's because it'll be easier to use `find usage` when you need to search for places where you had to do this.